### PR TITLE
Restructure README documentation

### LINKED ISSUE #62: Prepare doc for new 0.3.0 release
We need to update our README, because we did some changes which will destroy backward compatibility with `0.2.x`

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the pull request

Update doc for new upcoming release

`FloatingIp` flags to `EIP` flags (`ElasticIP`)

Resolves: #62 

## Acceptance tests result

```
=== RUN   TestDriver_SetConfigFromFlags
--- PASS: TestDriver_SetConfigFromFlags (0.00s)
=== RUN   TestDriver_Auth
--- PASS: TestDriver_Auth (3.88s)
=== RUN   TestDriver_Auth/default
    --- PASS: TestDriver_Auth/default (0.97s)
=== RUN   TestDriver_Auth/credentials
    --- PASS: TestDriver_Auth/credentials (0.98s)
=== RUN   TestDriver_Auth/ak/sk
    --- PASS: TestDriver_Auth/ak/sk (1.93s)
=== RUN   TestDriver_Create
--- PASS: TestDriver_Create (250.02s)
=== RUN   TestDriver_Create/ak/sk
Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/5045c215010c440d91b2f7dce1f3753b/os-floating-ips/], error message: {"message":"API not found","request_id":"7cff6d447085e74766d2617c9701d6e1"}

    --- PASS: TestDriver_Create/ak/sk (129.87s)
=== RUN   TestDriver_Create/default
Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/5045c215010c440d91b2f7dce1f3753b/os-floating-ips/], error message: {"message":"API not found","request_id":"23401558432df1c34a97db2f06ad3c05"}

    --- PASS: TestDriver_Create/default (120.15s)
=== RUN   TestDriver_Start
--- PASS: TestDriver_Start (215.54s)
=== RUN   TestDriver_CreateWithExistingSecGroups
Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/5045c215010c440d91b2f7dce1f3753b/os-floating-ips/], error message: {"message":"API not found","request_id":"c38a24ffc2e1b4a64c22d7ce97ca20a8"}

--- PASS: TestDriver_CreateWithExistingSecGroups (126.66s)
=== RUN   TestDriver_ExistingSSHKey
--- PASS: TestDriver_ExistingSSHKey (122.39s)
=== RUN   TestDriver_WithoutEIP
--- PASS: TestDriver_WithoutEIP (109.32s)
=== RUN   TestDriver_CreateWithUserData
Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/5045c215010c440d91b2f7dce1f3753b/os-floating-ips/], error message: {"message":"API not found","request_id":"7126d7d7fdb6ef375bfb0761b8b5a386"}

--- PASS: TestDriver_CreateWithUserData (130.78s)
=== RUN   TestDriver_UserDataRaw
--- PASS: TestDriver_UserDataRaw (1.45s)
=== RUN   TestDriver_ResolveServerGroup
--- PASS: TestDriver_ResolveServerGroup (6.41s)
=== RUN   TestDriver_FaultyRemove
--- PASS: TestDriver_FaultyRemove (8.83s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: docs/supported-options-v0.2.x.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+`--otc-availability-zone` | `OS_AVAILABILITY_ZONE` | eu-de-03                            | Availability zone
+`--otc-available-zone`    | `AVAILABLE_ZONE`       |                                     | Availability zone. **DEPRECATED**: use `-otc-availability-zone` instead
+`--otc-bandwidth-size`    | `BANDWIDTH_SIZE`       | 100 (MBit/s)                        | Bandwidth size
+`--otc-bandwidth-type`    | `BANDWIDTH_TYPE`       | PER (exclusive bandwidth)           | Bandwidth share type
+`--otc-cloud`             | `OS_CLOUD`             |                                     | Name of cloud in `clouds.yaml` file
+`--otc-domain-id`         | `OS_DOMAIN_ID`         |                                     | OpenTelekomCloud Domain ID
+`--otc-domain-name`       | `OS_DOMAIN_NAME`       |                                     | OpenTelekomCloud Domain name
+`--otc-elastic-ip`        | `ELASTIC_IP`           | 1                                   | If set to 0, elastic IP won't be created. **DEPRECATED**: use `-otc-skip-ip` instead
+`--otc-elastic-ip-type`   | `ELASTICIP_TYPE`       |                                     | Bandwidth type. **DEPRECATED!** Use `-otc-floating-ip-type` instead
```
Comment (anton-sidelnikov) [inline docs/supported-options-v0.2.x.md:16]: ELASTICIP_TYPE is ok, or should be ELASTIC_IP_TYPE?

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -8,46 +9,53 @@ OpenTelekomCloud driver for docker-machine
...
-AK/SK auth                                      | Yes           | Yes       | No
-Server group                                    | **Yes**       | No        | No
-Security group(s)                               | Multiple      | Single    | Multiple
-Prepared k8s security group                     | Yes           | No        | No
-Usage with rancher                              | Needs setup   | Built-in  | Built-in
+
+* [docker-machine-openstack](https://opendev.org/x/docker-machine-openstack) ― docker-machine built-in
+* [DockerMachine4OTC](https://github.com/Huawei/DockerMachineDriver4OTC) ― older OTC driver implementation by Huawei
+
+This driver versioned `v0.2.x` is inspired by `docker-machine-openstack` and targets to provide full backward compatibility
```
Comment (outcatcher) [inline README.md]: Both versions are inspired by `docker-machine-openstack`. But only `v0.2` aims for compatibility

File: README.md [hunk 3]
```diff
@@ -8,46 +9,53 @@ OpenTelekomCloud driver for docker-machine
...
+
+Feature                                        | OTC (new)   | OTC (old) | Openstack
+---                                            | ---         | ---       | ---
+Automated creation of required infrastructure  | **Yes**     | No        | No
+Support of `clouds.yaml` and `OS_CLOUD`        | **Yes**     | No        | No
+Support using of resource names instead of IDs | Yes         | No        | Yes
+Support of CI/CD pipelines                     | **Yes**     | No        | No
+User data injection                            | Yes         | No        | Yes
+Elastic IP pool selection                      | No          | No        | Yes
+Custom CA usage                                | No          | No        | Yes
```
Comment (outcatcher) [inline README.md]: `Custom CA` is supported in the new driver

File: README.md [hunk 4]
```diff
@@ -8,46 +9,53 @@ OpenTelekomCloud driver for docker-machine
...
+Usage with rancher                             | Needs setup | Built-in  | Built-in
 
 ### Installation
 
 Driver can be installed several ways
 
 #### From source code
-_(Requires Go 1.11+, gcc and make installed)_
+
+_(Requires Go 1.15+, gcc and make installed)_
```
Comment (outcatcher) [inline README.md]: Do we really require such a new version? What version is used in CI?

File: docs/usage-with-rancher.md [hunk 5]
```diff
@@ -1,20 +1,15 @@
...
 
-Currently, `docker-machine-opentelekomcloud` is not a part of the Rancher,
-so additional steps required to use it as a node driver
-
+Currently, `docker-machine-opentelekomcloud` is not a part of the Rancher, so additional steps required to use it as a node driver
 
 ### Remove old node driver:
 
- * Open Rancher UI page and go to `Tools` → `Drivers` → `Node Drivers`
+ * Open Rancher UI page and go to `Tools` -> `Drivers` -> `Node Drivers`
```
Comment (outcatcher) [inline docs/usage-with-rancher.md]: Why replace `→` with `->`?

File: README.md [hunk 6]
```diff
@@ -59,72 +67,31 @@ You will have to copy driver to directory in `PATH` so `docker-machine` would be
...
-`--otc-subnet-name`         | `SUBNET_NAME`             | subnet-docker-machine                 | Subnet name the machine will be connected on
-`--otc-token`               | `OS_TOKEN`                |                                       | Authorization token
-`--otc-tenant-id`           | `TENANT_ID`               |                                       | Project ID. DEPRECATED: use `-otc-project-id` instead
-`--otc-user-data-file`      | `OS_USER_DATA_FILE`       |                                       | File containing an userdata script
-`--otc-username`            | `OS_USERNAME`             |                                       | OpenTelekomCloud username
-`--otc-vpc-id`              | `VPC_ID`                  |                                       | VPC id the machine will be connected on
-`--otc-vpc-name`            | `OS_VPC_NAME`             | vpc-docker-machine                    | VPC name the machine will be connected on
+
+For versions `v0.3.x` see [supported-options](docs/supported-options-v0.3.x.md)
+For versions `v0.2.x` see [supported-options](docs/supported-options-v0.2.x.md)
```
Comment (outcatcher) [inline README.md]: Please add note that all old flags are supported for `v0.2.x`

File: docs/supported-options-v0.2.x.md (hunk 1 above)
Comment (outcatcher) [reply docs/supported-options-v0.2.x.md:16]: This is the old flag documentation

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (lego963) [reply README.md]: v1.13.5 for goreleaser

File: docs/usage-with-rancher.md (hunk 5 above)
Comment (lego963) [reply docs/usage-with-rancher.md]: I thought that it would be better insted of symbol. Should I revert it?

File: README.md [hunk 7]
```diff
@@ -8,46 +9,53 @@ OpenTelekomCloud driver for docker-machine
...
+convention.
+
+In versions `v0.3.+` duplicating options were removed and all environment variables are prefixed with `OS_`.
+
+Feature                                        | OTC (new)   | OTC (old) | Openstack
+---                                            | ---         | ---       | ---
+Automated creation of required infrastructure  | **Yes**     | No        | No
+Support of `clouds.yaml` and `OS_CLOUD`        | **Yes**     | No        | No
+Support using of resource names instead of IDs | Yes         | No        | Yes
+Support of CI/CD pipelines                     | **Yes**     | No        | No
```
Comment (outcatcher) [inline README.md]: What does `Support of CI/CD pipelines` mean?

File: README.md [hunk 8]
```diff
@@ -8,46 +9,53 @@ OpenTelekomCloud driver for docker-machine
...
+Elastic IP pool selection                      | No          | No        | Yes
+Custom CA usage                                | No          | No        | Yes
+Insecure mode (without TLS certificate check)  | No          | No        | Yes
+Bandwidth configuration                        | Yes         | Yes       | No
+Root volume configuration                      | Yes         | Yes       | No
+Optional usage of elastic IP                   | Yes         | Yes       | No
+AK/SK auth                                     | Yes         | Yes       | No
+Server group                                   | **Yes**     | No        | No
+Security group(s)                              | Multiple    | Single    | Multiple
+Usage with rancher                             | Needs setup | Built-in  | Built-in
```
Comment (outcatcher) [inline README.md]: `Usage with rancher` - better to call it "Rancher integration"

File: README.md (hunk 4 above)
Comment (outcatcher) [reply README.md]: Than `v1.13` should be enough

File: README.md (hunk 6 above)
Comment (outcatcher) [inline README.md]: Those two lines are concatenated in rendered markdown

File: docs/usage-with-rancher.md (hunk 5 above)
Comment (outcatcher) [reply docs/usage-with-rancher.md]: Originally it was used to look better than `->` 🙂

File: docs/supported-options-v0.2.x.md (hunk 1 above)
Comment (lego963) [reply docs/supported-options-v0.2.x.md:16]: yes, it's wrong, but in version `v0.2.4` we already had this typo

```
		mcnflag.StringFlag{
			Name:   "otc-elastic-ip-type",
			EnvVar: "ELASTICIP_TYPE",
			Usage:  "OpenTelekomCloud bandwidth type. DEPRECATED! Use -otc-floating-ip-type instead",
		},
```

File: README.md (hunk 7 above)
Comment (lego963) [reply README.md]: removed

File: README.md (hunk 8 above)
Comment (lego963) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (lego963) [reply README.md]: done

File: docs/usage-with-rancher.md (hunk 5 above)
Comment (lego963) [reply docs/usage-with-rancher.md]: done

**Review by anton-sidelnikov — DISMISSED** [review]

File: docs/usage-with-rancher.md [hunk 9]
```diff
@@ -1,20 +1,15 @@
...
+### Usage of new node driver:
 
  * Open OpenTelekomDriver [releases](https://github.com/opentelekomcloud/docker-machine-opentelekomcloud/releases) page and copy link of binary for 64-bit Linux
  * Click `Add New Driver` button, insert copied link and click `Create`
  * Wait for a while. Driver should be downloaded and be in `Active` state.
-
-## Create new template
-
-New OTC driver template supports all parameters supported by the old version
+ * Create new OTC driver template
```
Comment (outcatcher) [inline docs/usage-with-rancher.md]: Please add full stops at the end of each line

**Review by outcatcher — APPROVED** [review]

**Comment by lego963** [comment]
recheck

**Review by kucerakk — APPROVED** [review]

**Pull request merged**